webapp/channels/src/plugins/index.ts

```diff
@@ -18,13 +21,35 @@ import PluginRegistry from 'plugins/registry';
 import {ActionTypes} from 'utils/constants';
 import {getSiteURL} from 'utils/url';
 
+import type {GlobalState} from 'types/store';
+
 import {removeWebappPlugin} from './actions';
 
 // Including the fullscreen modal css to make it available to the plugins
 // (without lazy loading). This should be removed in the future whenever we
 // have all plugins migrated to common components that can be reused there.
 import 'components/widgets/modals/full_screen_modal.scss';
 
+interface Plugin {
+    initialize?: (registry: PluginRegistry, store: any) => void;
+    uninitialize?: () => void;
+
+    /**
+     * @deprecated Define an uninitialize method instead.
```

Comment:
Should we document the deprecation at https://developers.mattermost.com/integrate/reference/webapp/webapp-reference?

Reply:
That's a good idea, but that page doesn't actually even reference `deinitialize`. I forget how far back I got, but that method name was replaced years ago

Reply:
Actually, I found it referenced on another page of those docs. I'll remove that reference since there's no reason for people to use `deinitialize` instead of `uninitialize`